src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/Passes/CSharpOnTypeFormattingPass.cs

```diff
@@ -119,7 +119,7 @@ protected async override Task<ImmutableArray<TextChange>> ExecuteCoreAsync(Forma
         var cleanedText = formattedText.WithChanges(cleanupChanges);
         _logger.LogTestOnly($"After CleanupDocument:\r\n{cleanedText}");
 
-        changedContext = await changedContext.WithTextAsync(cleanedText).ConfigureAwait(false);
+        changedContext = await changedContext.WithTextAsync(cleanedText, cancellationToken).ConfigureAwait(false);
 
         cancellationToken.ThrowIfCancellationRequested();
```

Comment:
nit: do we need this throw now that theoretically `WithTextAsync` would have thrown?

Reply:
Nope. I missed that one. Good catch! I'll remove the throw.

Reply:
But it _would_ throw. :smile:

At some point, you need to trust that something is going to `ThrowIfCancellationRequested` or return `Task.FromCanceled<T>(...)`.